src/gpa/activateGPA.js

```diff
@@ -2,102 +2,95 @@
+			append_checkbox($(this).children('.col1'), false);
+			return;
+		}
+		is_checked = true;
+		type = $(this).children('.col5').html().trim().slice(6);
+		grade = $(this).children('.col8').html().trim().slice(6);
+		console.log(grade, grade.length);
+		if (exclude_list.indexOf(type) > -1 || grade == '' || grade == 'I') is_checked = false;
+		if (is_checked) courses_checked.add(course_id);
+		append_checkbox($(this).children('.col1'), is_checked);
+	});
+};
 
-show_total_gpa = function(){
-    var courses = [];
-    $('#gpa_button').val('Calculating');
-    $('#gpa_bar').remove();
-    total_grades = 0;
-    total_credits = 0;
-    if ($(".cgpa_cal_check").length==0)
-        add_checkboxes();
-    elems = $(".hierarchyLi.dataLi").not(".hierarchyHdr, .hierarchySubHdr");
-    var type_credits_map = new Map();
-    elems.each(function(i){
-        if ($(this).find(".cgpa_cal_check:checked").length == 0 )
-            return;
+show_total_gpa = function () {
+	var courses = [];
+	$('#gpa_button').val('Calculating');
+	$('#gpa_bar').remove();
+	total_grades = 0;
+	total_credits = 0;
+	if ($('.cgpa_cal_check').length == 0) add_checkboxes();
+	elems = $('.hierarchyLi.dataLi').not('.hierarchyHdr, .hierarchySubHdr');
+	var type_credits_map = new Map();
+	elems.each(function (i) {
+		if ($(this).find('.cgpa_cal_check:checked').length == 0) return;
 
-        var course = new Object();
-        course.code = $(this).children(".col1").contents().filter(function(){
-            return this.nodeType == Node.TEXT_NODE;
-          }).text().trim();
-        course.name = $(this).children(".col2").html().trim();
-        course.type = $(this).children(".col5").html().trim().slice(6);
-        course.grade = $(this).children(".col8").html().trim().slice(6);
-        course.credits = Number($(this).children(".col3").html().trim().slice(6));
-        if (!(course.grade in grade_values))
-            return;
+		var course = new Object();
+		course.code = $(this)
+			.children('.col1')
+			.contents()
+			.filter(function () {
+				return this.nodeType == Node.TEXT_NODE;
+			})
+			.text()
+			.trim();
+		course.name = $(this).children('.col2').html().trim();
+		course.type = $(this).children('.col5').html().trim().slice(6);
+		course.grade = $(this).children('.col8').html().trim().slice(6);
+		course.credits = Number($(this).children('.col3').html().trim().slice(6));
+		if (!(course.grade in grade_values)) return;
 
-        if (type_credits_map.has(course.type))
-          type_credits_map.set(course.type, type_credits_map.get(course.type) + course.credits);
-        else
-          type_credits_map.set(course.type, course.credits);
+		if (type_credits_map.has(course.type))
+			type_credits_map.set(course.type, type_credits_map.get(course.type) + course.credits);
+		else type_credits_map.set(course.type, course.credits);
 
+		grade_value = grade_values[course.grade];
+		credits = (course.grade === 'S' || course.type === 'Additional') ? 0 : course.credits;
```

Comment:
@dark-trojan Good work on the PR! Can you please document exactly what's being done here?